lib/RTCSession.js

```diff
@@ -353,6 +353,17 @@ module.exports = class RTCSession extends EventEmitter
       extraHeaders.push(`P-Preferred-Identity: ${this._ua.configuration.uri.toString()}`);
       extraHeaders.push('Privacy: id');
     }
+    else
+    {
+      if (options.from_uri)
+      {
+        requestParams.from_uri = URI.parse(options.from_uri);
```

Comment:
`P-Preferred-Identity` is required if URI in From header does not honor the original one.

```suggestion
        requestParams.from_uri = URI.parse(options.from_uri);
        extraHeaders.push(`P-Preferred-Identity: ${this._ua.configuration.uri.toString()}`);
```